app/gui2/src/util/ast/abstract.ts

```diff
@@ -967,6 +984,13 @@ export class BodyBlock extends Ast {
     this.lines_ = lines
   }
 
+  addTrailingNewline(module: MutableModule) {
+    new BodyBlock(module, this.exprId, [
+      ...this.lines_,
+      { newline: { node: Token.new('\n') }, expression: null },
+    ])
+  }
```

Comment:
It seems wrong to me that this function doesn't return anything, and also doesn't modify the block in-place. What is the effect of calling this twice on the same body block object?